grpc/src/main/java/com/linecorp/armeria/client/grpc/GrpcClientBuilder.java

```diff
@@ -0,0 +1,498 @@
+     *   <li><a href="https://scalapb.github.io/">ScalaPB</a></li>
+     * </ul>
+     */
+    public GrpcClientBuilder clientStubFactory(GrpcClientStubFactory clientStubFactory) {
+        requireNonNull(clientStubFactory, "clientStubFactory");
+        return option(GRPC_CLIENT_STUB_FACTORY.newValue(clientStubFactory));
+    }
+
+    /**
+     * Adds the {@link ClientInterceptor}s to the gRPC client stub.
+     * The specified interceptor(s) is/are executed in reverse order.
+     */
+    public GrpcClientBuilder intercept(ClientInterceptor... interceptors) {
+        requireNonNull(interceptors, "interceptors");
+        return intercept(ImmutableList.copyOf(interceptors));
+    }
+
+    /**
+     * Adds the {@link ClientInterceptor}s to the gRPC client stub.
+     * The specified interceptor(s) is/are executed in reverse order.
+     */
+    public GrpcClientBuilder intercept(Iterable<? extends ClientInterceptor> interceptors) {
+        requireNonNull(interceptors, "interceptors");
+        this.interceptors.addAll(interceptors);
+        return this;
+    }
+
+    /**
+     * Unsupported operation. {@code rpcDecorator} only supports Thrift.
+     * @deprecated Use either {@link #decorator(DecoratingHttpClientFunction)} or
+     *             {@link #intercept(ClientInterceptor...)} instead.
+     */
+    @Deprecated
+    @Override
+    public GrpcClientBuilder rpcDecorator(Function<? super RpcClient, ? extends RpcClient> decorator) {
+        throw new UnsupportedOperationException("rpcDecorator() does not support gRPC. " +
+                                                "Use either decorator() or intercept()");
+    }
+
+    /**
+     * Unsupported operation. {@code rpcDecorator} only supports Thrift.
+     * @deprecated Use either {@link #decorator(DecoratingHttpClientFunction)} or
+     *             {@link #intercept(ClientInterceptor...)} instead.
+     */
+    @Deprecated
+    @Override
+    public GrpcClientBuilder rpcDecorator(DecoratingRpcClientFunction decorator) {
+        throw new UnsupportedOperationException("rpcDecorator() does not support gRPC. " +
+                                                "Use either decorator() or intercept()");
+    }
+
+    /**
+     * Returns a newly-created gRPC client which implements the specified {@code clientType}, based on the
+     * properties of this builder.
+     */
+    public <T> T build(Class<T> clientType) {
+        requireNonNull(clientType, "clientType");
+
+        final List<ClientInterceptor> clientInterceptors = interceptors.build();
+        if (!clientInterceptors.isEmpty()) {
+            option(GrpcClientOptions.INTERCEPTORS.newValue(clientInterceptors));
+        }
+
+        final Object client;
+        final ClientOptions options = buildOptions();
+        final ClientFactory factory = options.factory();
+        URI uri = this.uri;
+        if (uri != null) {
+            if (schemeChanged) {
```

Comment:
Shouldn't it better to set a new URI if `this.uri` is not `null` when the scheme is set?